BedrockServer.cpp

```diff
@@ -2258,90 +2235,27 @@ void BedrockServer::handleSocket(Socket* s, bool isControl) {
             } else {
                 // If we weren't able to deserialize a complete request, and we're shutting down, give up.
                 if (_shutdownState != RUNNING && _lastChance && _lastChance < STimeNow()) {
-                    SINFO("Closing socket " << s->id << " with incomplete data and no pending command: shutting down.");
+                    SINFO("Closing socket " << socket.id << " with incomplete data and no pending command: shutting down.");
                 }
             }
-        } else if (s->state == STCPManager::Socket::SHUTTINGDOWN || s->state == STCPManager::Socket::CLOSED) {
+        } else if (socket.state == STCPManager::Socket::SHUTTINGDOWN || socket.state == STCPManager::Socket::CLOSED) {
             // Do nothing here except prevent the warning below from firing. This loop should exit on the next
             // iteration.
         } else {
-            SWARN("Socket in unhandled state: " << s->state);
+            SWARN("Socket in unhandled state: " << socket.state);
         }
     }
 
     // At this point out socket is closed and we can clean up.
     // Note that we never return early, we always want to hit this code and decrement our counter and clean up our socket.
     _outstandingSocketThreads--;
-    delete s;
```

Comment:
Because we pass by move, our socket will automatically get destroyed when we return from this function, sowe don't need to do this manually.